### PR TITLE
Add route for the new profile mini-app

### PR DESCRIPTION
**Pull request by acelaya**
This adds preliminary support for a new mini-app handled by `client`, which purpose will be to manage the user profile settings, and eventually replace the server-side handled `/account/settings` route.

Refs: #7820

> I'm creating this as a draft, as I'm adding the changes to client in parallel, and it's likely that I need to push some extra changes.

### CODE REVIEW

File: h/routes.py
Comment (acelaya) [inline h/routes.py:184]: We discussed using `/profile` as the path. However, I have found this path exists in `h` since forever, and it is configured to redirect to `/account/settings`.

We should decide if this redirect is still relevant and remove it if it isn't, or just use a different path.

File: h/views/client.py
Comment (acelaya) [inline h/views/client.py:47]: This `@view_config(..)` call is the same as for notebook and sidebar, only `route_name` is different. I need to check if there's some way to not triplicate the code.

**Comment by lyzadanger** [comment]
@acelaya Would you be so kind as to split this into two commits and two PRs? These items are related to each other but are independent. Let's get the feature flag landed first...

**Comment by acelaya** [comment]
> @acelaya Would you be so kind as to split this into two commits and two PRs? These items are related to each other but are independent. Let's get the feature flag landed first...

Absolutely :)

**Review by lyzadanger — COMMENTED** [review]
There's a bit of a chicken-and-egg problem here. 

I think this PR is fine, but I wonder if we might want to land it _after_ landing client support for the route in question. The problem is that without client-side routing, the experience at `/user-profile` (e.g. locally at `localhost:5000/user-profile`) is broken. It's trying to render the Sidebar (because that's the routing fallback, I'd imagine) and is in an eternal half-loaded state because it can't find its host frame. 

Let's see if we might get the basic bits of the client-side stuff sorted out? What are your thoughts?

**Comment by acelaya** [comment]
Yeah, works for me.

File: h/routes.py
Comment (robertknight) [reply h/routes.py:184]: > We should decide if this redirect is still relevant and remove it if it isn't, or just use a different path.

Checking the git history in h provides some insight into how likely it is that anyone is relying on the old views. The original `/profile` view was removed in a0d0eba3658da680ed1b446d3b246450061f50e0 (Oct 2015) with the comment:

> Remove standalone profile/edit_profile/disable_user views
> These views are broken, and I can't find any evidence that they ever actually worked.

On that basis I think we should feel free to use `/profile` as a URL for new functionality, though we might want to use a temporary URL while we develop the app. Something that occurred to me while looking at the new app is that I think of "profile" and "account settings" as different things in a social application:

- The user's profile is the identity they display to the outside world
- The user's settings includes things like contact methods (phone number, email) notifications, privacy, API keys etc. which affect their experience but not how others see them

We should discuss the routes and naming in more detail this week.

Comment (acelaya) [reply h/routes.py:184]: > Checking the git history in h provides some insight into how likely it is that anyone is relying on the old views. The original /profile view was removed in https://github.com/hypothesis/h/commit/a0d0eba3658da680ed1b446d3b246450061f50e0 (Oct 2015) with the comment:

I'm thinking that we could even go a bit further, and check the logs to see if there's any request hitting this endpoint at all.

Comment (robertknight) [reply h/routes.py:184]: > I'm thinking that we could even go a bit further, and check the logs to see if there's any request hitting this endpoint at all.

This redirect is served by a Pyramid "tweet" in h, which means that the redirect should definitely be visible in h's access logs, and I think redirects will be visible as transactions in New Relic, but I'm not certain. h's access logs I believe are available in Papertrail, but check with Ian on that.

Comment (lyzadanger) [reply h/routes.py:184]: > This redirect is served by a Pyramid "tweet" in h, which means that the redirect should definitely be visible in h's access logs, and I think redirects will be visible as transactions in New Relic, but I'm not certain. h's access logs I believe are available in Papertrail, but check with Ian on that.

Here I swoop in pedantically, but I think, to ease confusion, it's important to point out that Rob means _tween_. GRIN.

Comment (robertknight) [reply h/routes.py:184]: Yes, indeed. "tween" is short for "between" and IIRC it is similar to "middleware" in other frameworks.

File: h/routes.py
Comment (robertknight) [inline h/routes.py:184]: Can you add a link to the relevant GitHub issue, assuming there is one.

**Review by robertknight — APPROVED** [review]
LGTM, but add a GitHub issue link if there is a suitable issue.

File: h/routes.py
Comment (acelaya) [reply h/routes.py:184]: Done

**Pull request merged**
